### PR TITLE
Store refactor

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
This PR does a few smaller refactors.

1. add soft-deletion and update/creation/deletion times to the db by adding the corresponding columns to the database tables.
2. Refactor the database types into a single one that simply implements the corresponding interfaces.
3. Moves the methods for the hostsets into a separate file. That way every interface gets its own file.

### CODE REVIEW

File: internal/stores/sql.go [hunk 1]
```diff
@@ -19,3 +45,65 @@ func NewEphemeralSQLiteConnection(name string) gorm.Dialector {
 func NewSQLiteConnection(path string) gorm.Dialector {
 	return sqlite.Open(path)
 }
+
+// NewSQLHostDB uses a given Dialector to connect to a SQL database.  NOTE: Only
```
Comment (peterjan) [inline internal/stores/sql.go:49]: embarrassing

File: internal/stores/sql_test.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package stores
+
+import (
+	"encoding/hex"
+
+	"go.sia.tech/siad/modules"
+	"lukechampine.com/frand"
+)
+
+// newTestSQLContractStore creates a new SQLContractStore for testing.
```
Comment (peterjan) [inline internal/stores/sql_test.go:10]:
```suggestion
// newTestSQLStore creates a new SQLStore for testing.
```

**Review by peterjan — APPROVED** [review]

**Comment by peterjan** [comment]
nice little refactor 👍 also great we're adding soft deletion here I think that's a good call

File: internal/stores/sql.go [hunk 3]
```diff
@@ -2,11 +2,37 @@ package stores
...
+		db *gorm.DB
+	}
+)
+
+// Check that the SQLStore implements all the required interfaces.
+var (
+	_ bus.ContractStore = &SQLStore{}
+	_ bus.HostDB        = &SQLStore{}
+	_ bus.HostSetStore  = &SQLStore{}
+	_ bus.ObjectStore   = &SQLStore{}
```
Comment (lukechampine) [inline internal/stores/sql.go:33]: perhaps we should merge these into a single `bus.Store` 🤔

**Pull request merged**